manager/state/raft/raft.go

```diff
@@ -269,6 +271,10 @@ func (n *Node) JoinAndStart(ctx context.Context) (err error) {
 	n.confState = snapshot.Metadata.ConfState
 	n.appliedIndex = snapshot.Metadata.Index
 	n.snapshotIndex = snapshot.Metadata.Index
+	n.writtenIndex, err = n.raftStore.LastIndex()
+	if err != nil {
+		panic("could not get last index out of raft store")
+	}
```

Comment:
This is fine as is, but just as an aside, `LastIndex` always returns a nil error. It really shouldn't return an error value.